server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -308,6 +309,18 @@ protected static int getMaxConcurrencyWriting(
   public StatusCode cacheShuffleData(
       String appId, int shuffleId, boolean isPreAllocated, ShufflePartitionedData spd) {
     refreshAppId(appId);
+    long partitionSize = getPartitionDataSize(appId, shuffleId, spd.getPartitionId());
+    long deltaSize = spd.getTotalBlockSize();
+    partitionSize += deltaSize;
+    if (shuffleBufferManager.isPartitionExceededMaxSize(partitionSize)) {
+      throw new ExceedPartitionSizeLimitException(
+          "Current partition size: "
+              + partitionSize
+              + " exceeded the max size: "
+              + shuffleBufferManager.getPartitionMaxSize()
+              + " if cache this shuffle data with size: "
```

Comment:
`before caching` will be better?

Because the other part of the code happens to be `after caching`.